pandas/_libs/tslibs/ccalendar.pyi

```diff
@@ -0,0 +1,13 @@
+
+DAYS: list[str]
```

Comment:
how does `list[str]` work in Python < 3.9? (asking because I don't know)

Reply:
i decided to try this out bc i think ive seen @simonjayhawkins using this pattern using it recently

Reply:
> how does `list[str]` work in Python < 3.9? (asking because I don't know)

pyi files don't go through the Python interpreter. so have the advantage of being able to use the latest typing features understood by mypy/pyre/pyright/pytype. `list[str]` is PEP 585 compliant.

---

side note: we are in the process of attempting to inline the microsoft pyright stubs. inline types not only need to satisfy the python interpreter supported (i.e. python 3.7 **without** typing extensions) but the code in the annotated functions also needs to be consistent with the in-line function annotations.

so (hopefully without sounding too negative) I imagine that not only will that take a while but may also not even be possible in some cases.

For type checking user code against the public api, I expect using stubs (eg. bundled with pyright or 3rd party such as https://github.com/predictive-analytics-lab/data-science-types (development ceased 16 Feb 2021)) will be the better option for the foreseeable future.

However, internal consistency/robustness is achieved though adding inline types and checking with mypy (which has excellent features that support the gradual typing process.) 

---

we do yet have a good way of generating type stubs from cython files, so in order to have types available for the compiled code, we probably need to manually curate these. So being able to use the pyright stubs here would be beneficial. 

@jbrockmendel Are these stubs manually curated or forked from pyright? I believe @Dr-Irv is coordinating between pandas and pyright and may have some input on this.

Reply:
> Are these stubs manually curated or forked from pyright?

Manually curated

Reply:
I think doing this is OK, as adding stubs may reveal problems in the python code that may need to be fixed. So doing this before pulling in the pyright stubs could be advantageous and make the migration smoother.

Reply:
> @jbrockmendel Are these stubs manually curated or forked from pyright? I believe @Dr-Irv is coordinating between pandas and pyright and may have some input on this.

I wouldn't say I'm "coordinating", but the work that has been done by the Microsoft team provides a good starting point (and possible reference) for pandas type stubs.  See https://github.com/microsoft/python-type-stubs/tree/main/pandas